src/test/java/uk/ac/cam/cl/dtg/isaac/api/AssignmentFacadeIT.java

```diff
@@ -432,4 +437,35 @@ public void assignBulkEndpoint_scheduleSingleAssignmentWithScheduledDateAfterDue
         // check status code is NO_CONTENT (successful)
         assertEquals(Response.Status.NO_CONTENT.getStatusCode(), deleteAssignmentResponse.getStatus());
     }
+
+    @Test public void getAssignmentProgressDownloadCSV_succeeds() throws Exception {
+        // log in as Test teacher, create request
+        LoginResult teacherLogin = loginAs(httpSession, ITConstants.TEST_TEACHER_EMAIL,
+                ITConstants.TEST_TEACHER_PASSWORD);
+        HttpServletRequest downloadAssignmentRequest = createRequestWithCookies(new Cookie[]{teacherLogin.cookie});
+        replay(downloadAssignmentRequest);
+
+        Response downloadAssignmentResponse = assignmentFacade.getAssignmentProgressDownloadCSV(downloadAssignmentRequest, 4L, "excel");
+        String downloadAssignmentContents = downloadAssignmentResponse.getEntity().toString();
+        
+        // Note that the first two lines have a trailing whitespace. If your IDE automatically removes these from the expected file it may cause a mismatch
+        try (FileInputStream expectedFile = new FileInputStream("src/test/resources/expected_assignment_export.csv");) {
+            String expectedContents = IOUtils.toString(expectedFile, (Charset) null);
+
+            assertEquals(expectedContents, downloadAssignmentContents);
+        }
+    }
+
+    @Test public void getAssignmentProgressDownloadCSV_permissionDenied() throws Exception {
+        // log in as Test teacher, create request
+        LoginResult teacherLogin = loginAs(httpSession, ITConstants.TEST_TUTOR_EMAIL,
+                ITConstants.TEST_TUTOR_PASSWORD);
+        HttpServletRequest downloadAssignmentRequest = createRequestWithCookies(new Cookie[]{teacherLogin.cookie});
+        replay(downloadAssignmentRequest);
+
+        Response downloadAssignmentResponse = assignmentFacade.getAssignmentProgressDownloadCSV(downloadAssignmentRequest, 4L, "excel");
+        String downloadAssignmentContents = downloadAssignmentResponse.getEntity().toString();
+
+        assertEquals("Error Code: 403 Error Message: 403 \nnull", downloadAssignmentContents);
```

Comment:
Instead of this, couldn't you just check that `downloadAssignmentResponse.getStatus()` is `403`? You don't really care about the message, you only care about the status code, right?